forwardbot_gitlab/models/merge_request.py

```diff
@@ -0,0 +1,276 @@
+        self.ensure_one()
+        self.validate_object_attributes(object_attributes)
+        self.write(
+            {
+                "name": object_attributes["title"],
+                "state": object_attributes["state"],
+                "target_branch": object_attributes["target_branch"],
+                "processed": False,
+            }
+        )
+
+    @api.model
+    def validate_object_attributes(self, object_attributes: Dict):
+        for field in self.required_merge_request_fields:
+            if not object_attributes.get(field):
+                raise ValueError(f"missing field object_attributes.{field}")
+
+    @api.model
+    def obtain_from_event(self, data: Dict) -> Tuple["MergeRequest", bool]:
+        """Obtain (search and create if not found) a record based on the data
+        provided by a Merge Request event as described by Gitlab on:
+        https://docs.gitlab.com/ee/user/project/integrations/webhook_events.html#merge-request-events
+        """
+        data_object = data.get("object_attributes", {})
+        self.validate_object_attributes(data_object)
+
+        merge_request = self.search([("global_id", "=", data_object["id"])])
+        if merge_request:
+            return merge_request, False
+
+        target_project_id = data_object["target_project_id"]
+        target_project = self.env["forwardbot_gitlab.project"].search([("remote_id", "=", target_project_id)], limit=1)
+        if not target_project:
+            target_project = self.env["forwardbot_gitlab.project"].create({"remote_id": target_project_id})
+
+        return (
+            self.env["forwardbot_gitlab.merge.request"].create(
+                {
+                    "name": data_object["title"],
+                    "source_branch": data_object["source_branch"],
+                    "source_project_id": data_object["source_project_id"],
+                    "target_branch": data_object["target_branch"],
+                    "target_project_id": target_project.id,
+                    "global_id": data_object["id"],
+                    "project_iid": data_object["iid"],
+                    "state": data_object["state"],
+                }
+            ),
+            True,
+        )
+
+    @api.model
+    def prune_merge_requests(self, auto_commit=False, limit=50, timeout=10):
+        url, token = get_api_data(self.env)
+        if not url or not token:
+            return
+
+        # TODO: Rotate MRs based on last checked
+        merge_requests: MergeRequest = self.search([("state", "!=", "merged")], limit=limit)
+        _logger.info("Reviewing %d merge requests for pruning", len(merge_requests))
+        if not merge_requests:
+            return
+
+        pruned_mrs = self.browse([])
+        client = ForwardbotClient(url, token, timeout)
+        for merge_request in merge_requests:
+            if merge_request.target_branch in merge_request.target_project_id.protected_branches:
```

Comment:
I think not all the stable branches for odoo-version are configured as protected_branches

and not all the protected_branches could be related for an odoo-version

Even if it should be protected_branches we should be able to configure the sticky branches similar to runbot.branch

Even if it could be auto-configured using this way

Note: I'm not sure about my own comment

But we need to discuss it in order to have better context

Reply:
It was the only API way I could think to filter out branches, since I found no reference on the REST API. I visualized this a zero config setup, but I have added an option to overwrite branches and add your own.